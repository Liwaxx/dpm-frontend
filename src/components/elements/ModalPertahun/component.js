import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import Modal from '@material-ui/core/Modal';
import Month from '../Month';
import { IMAGES } from '../../../configs';

export default class Component extends React.Component {
  state = {
    open: false,
  };

  handleOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { classes } = this.props;
    return (
      <div className="modal">
        <Button id="modal-pertahun-open" onClick={this.handleOpen} className={classes.text}>Oktober 2018</Button>
        <Modal open={this.state.open}>
          <div className={classes.paper}>
            <div className={classes.header}>
              <Month />
              <Button id="modal-pertahun-close" onClick={this.handleClose} className={classes.close}>
                <img src={IMAGES.CLOSE} className={classes.closeImage} />
              </Button>
            </div>
            <div className={classes.line1}>
              <Button> Jan </Button>
              <Button> Feb </Button>
              <Button> Mar </Button>
              <Button> Apr </Button>
            </div>
            <div className={classes.line2}>
              <Button> Mei </Button>
              <Button> Jun </Button>
              <Button> Jul </Button>
              <Button> Aug </Button>
            </div>
            <div className={classes.line3}>
              <Button> Sep </Button>
              <Button> Okt </Button>
              <Button> Nop </Button>
              <Button> Des </Button>
            </div>
          </div>
        </Modal>
      </div>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object.isRequired,
};
