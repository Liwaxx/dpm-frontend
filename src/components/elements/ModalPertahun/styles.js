const styles = ({
  paper: {
    marginLeft: '25%',
    marginTop: '5%',
    backgroundColor: '#FFFFFF',
    boxShadow: '0 2px 10px 0 rgba(0, 0, 0, 0.07999999821186066)',
    width: '260px',
    height: '160px'
  },
  header: {
    marginLeft: '8px',
  },
  close: {
    marginLeft: '205px',
    marginTop: '-100px',
    width: '20px',
    height: '20px',
  },
  closeImage: {
    width: '20px',
  },
  line1: {
    marginLeft: '0px',
    marginTop: '-5px',
  },
  line2: {
    marginLeft: '0px',
    marginTop: '-5px',
  },
  line3: {
    marginLeft: '0px',
    marginTop: '-5px',
  },
  text: {
    color: '#53B9EC',
  }
});
export default styles;

