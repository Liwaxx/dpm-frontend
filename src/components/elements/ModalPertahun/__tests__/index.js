import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ModalPertahun from '../index';
import Component from '../component';

jest.mock('../../Month');
jest.mock('../../../../configs');

describe('ModalPertahun', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ModalPertahun />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('opens modal when the button clicked', () => {
    const wrapper = shallow(<Component classes={{}} />);
    wrapper.find('#modal-pertahun-open').first().simulate('click');
    expect(wrapper.state('open')).toEqual(true);
  });

  it('closes modal when the button clicked', () => {
    const wrapper = shallow(<Component classes={{}} />);
    wrapper.setState({ open: true });
    wrapper.find('#modal-pertahun-close').first().simulate('click');
    expect(wrapper.state('open')).toEqual(false);
  });
});