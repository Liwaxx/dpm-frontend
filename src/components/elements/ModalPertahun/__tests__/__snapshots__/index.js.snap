// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPertahun renders correctly 1`] = `
<div
  className="modal"
>
  <button
    className="MuiButtonBase-root-35 MuiButton-root-9 MuiButton-text-11 MuiButton-flat-14 Component-text-8"
    disabled={false}
    id="modal-pertahun-open"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-10"
    >
      Oktober 2018
    </span>
    <span
      className="MuiTouchRipple-root-38"
    />
  </button>
</div>
`;
